script.service.hue/resources/lib/kodiHue.py

```diff
@@ -0,0 +1,498 @@
+                             transitionTime) * 10)  # Hue API transition time is in 100msec. *10 to convert to seconds.
+            logger.debug("In kodiHue createHueScene. Res: {}".format(res))
+            if res[0]["success"]:
+                xbmcgui.Dialog().ok(heading=_("Create New Scene"),message=_("Scene successfully created![CR]You may now assign your Scene to player actions."))
+            #   xbmcgui.Dialog().notification(_("Hue Service"), _("Scene Created"))
+            else:
+                xbmcgui.Dialog().ok(_("Error"), _("Error: Scene not created."))
+
+
+def deleteHueScene(bridge):
+    logger.debug("In kodiHue deleteHueScene")
+    scene = selectHueScene(bridge)
+    if scene is not None:
+        confirm = xbmcgui.Dialog().yesno(heading=_("Delete Hue Scene"), message=_("Are you sure you want to delete this scene:[CR]" + str(scene[1])))
+    if scene and confirm:
+        scenes = bridge.scenes
+        res = scenes[scene[0]](http_method='delete')
+        logger.debug("In kodiHue createHueGroup. Res: {}".format(res))
+        if res[0]["success"]:
+            xbmcgui.Dialog().notification(_("Hue Service"), _("Scene deleted"))
+        else:
+            xbmcgui.Dialog().notification(_("Hue Service"), _("ERROR: Scene not created"))
+
+
+def _discoverNupnp():
+    logger.debug("In kodiHue discover_nupnp()")
+    try:
+        # ssl chain on new URL seems to be fixed
+        # req = requests.get('https://www.meethue.com/api/nupnp')
+        req = requests.get('https://discovery.meethue.com/')
+    except requests.exceptions.ConnectionError as e:
+        logger.info("Nupnp failed: {}".format(e))
+        return None
+
+    res = req.json()
+    bridge_ip = None
+    if res:
+        bridge_ip = res[0]["internalipaddress"]
+    return bridge_ip
+
+
+def _discoverSsdp():
+    from . import ssdp
+    from urllib.parse import urlsplit
+
+    try:
+        ssdp_list = ssdp.discover("upnp:rootdevice", timeout=10, mx=5)
+    except Exception as exc:
+        logger.exception("SSDP error: {}".format(exc.args))
+        xbmcgui.Dialog().notification(_("Hue Service"), _("Network not ready"), xbmcgui.NOTIFICATION_ERROR)
+        return None
+
+    logger.debug("ssdp_list: {}".format(ssdp_list))
+
+    bridges = [u for u in ssdp_list if 'IpBridge' in u.server]
+    if bridges:
+        ip = urlsplit(bridges[0].location).hostname
+        logger.debug("ip: {}".format(ip))
+        return ip
+    return None
+
+
+def bridgeDiscover(monitor):
+    logger.debug("Start bridgeDiscover")
+    # Create new config if none exists. Returns success or fail as bool
+    ADDON.setSettingString("bridgeIP", "")
+    ADDON.setSettingString("bridgeUser", "")
+    settings_storage['connected'] = False
+
+    progressBar = xbmcgui.DialogProgress()
+    progressBar.create(_('Searching for bridge...'))
+    progressBar.update(5, _("Discovery started"))
+
+    complete = False
+    while not progressBar.iscanceled() and not complete:

```

Comment:
please include a `not monitor.abortRequested()` check

Reply:
Good catch, will be corrected